recipes/embree3/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+        "geometry_quad": True,
+        "geometry_subdivision": True,
+        "geometry_triangle": True,
+        "geometry_user": True,
+        "ray_packets": True,
+        "ray_masking": False,
+        "backface_culling": False,
+        "ignore_invalid_rays": False,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+    
+    def configure(self):
+        if self.settings.compiler == "apple-clang" and self.settings.compiler.version == "9.1":
+            self.options.shared = True
+
+        version = tools.Version(self.settings.compiler.version)
+        if self.settings.compiler == "clang" and version < "4":
+            raise ConanInvalidConfiguration("Clang < 4 is not supported")
+        elif self.settings.compiler == "Visual Studio" and version < "15":
+            raise ConanInvalidConfiguration("Visual Studio < 15 is not supported")
+
+        if self._is_linux_clang_libcxx:
+            raise ConanInvalidConfiguration("conan recipe for Embree v{0} \
+                cannot be built with clang libc++, use libstdc++ instead".format(self.version))
+
+        elif self._is_mac_apple_clang_static:
+            raise ConanInvalidConfiguration("Embree cannot be built into static library with apple-clang,\
+                try shared library instead")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        target_name = "{0}-{1}".format("embree", self.version)
+        os.rename(target_name, self._source_folder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        # Configure CMake library build:
+        cmake.definitions["EMBREE_STATIC_LIB"] = not self.options.shared
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.definitions["EMBREE_TUTORIALS"] = False
+        cmake.definitions["EMBREE_GEOMETRY_CURVE"] = self.options.geometry_curve
+        cmake.definitions["EMBREE_GEOMETRY_GRID"] = self.options.geometry_grid
+        cmake.definitions["EMBREE_GEOMETRY_INSTANCE"] = self.options.geometry_instance
+        cmake.definitions["EMBREE_GEOMETRY_QUAD"] = self.options.geometry_quad
+        cmake.definitions["EMBREE_GEOMETRY_SUBDIVISION"] = self.options.geometry_subdivision
+        cmake.definitions["EMBREE_GEOMETRY_TRIANGLE"] = self.options.geometry_triangle
+        cmake.definitions["EMBREE_GEOMETRY_USER"] = self.options.geometry_user
+        cmake.definitions["EMBREE_RAY_PACKETS"] = self.options.ray_packets
+        cmake.definitions["EMBREE_RAY_MASK"] = self.options.ray_masking
+        cmake.definitions["EMBREE_BACKFACE_CULLING"] = self.options.backface_culling
+        cmake.definitions["EMBREE_IGNORE_INVALID_RAYS"] = self.options.ignore_invalid_rays
+        cmake.definitions["EMBREE_ISPC_SUPPORT"] = False
+        cmake.definitions["EMBREE_TASKING_SYSTEM"] = "INTERNAL"
+
+        cmake.configure(build_folder=self._build_folder)
+        return cmake
```

Comment:
Re-use cmake instance, otherwise it will run cmake configure twice, one for build, one for package.

```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)

        # Configure CMake library build:
        self._cmake.definitions["EMBREE_STATIC_LIB"] = not self.options.shared
        self._cmake.definitions["BUILD_TESTING"] = False
        self._cmake.definitions["EMBREE_TUTORIALS"] = False
        self._cmake.definitions["EMBREE_GEOMETRY_CURVE"] = self.options.geometry_curve
        self._cmake.definitions["EMBREE_GEOMETRY_GRID"] = self.options.geometry_grid
        self._cmake.definitions["EMBREE_GEOMETRY_INSTANCE"] = self.options.geometry_instance
        self._cmake.definitions["EMBREE_GEOMETRY_QUAD"] = self.options.geometry_quad
        self._cmake.definitions["EMBREE_GEOMETRY_SUBDIVISION"] = self.options.geometry_subdivision
        self._cmake.definitions["EMBREE_GEOMETRY_TRIANGLE"] = self.options.geometry_triangle
        self._cmake.definitions["EMBREE_GEOMETRY_USER"] = self.options.geometry_user
        self._cmake.definitions["EMBREE_RAY_PACKETS"] = self.options.ray_packets
        self._cmake.definitions["EMBREE_RAY_MASK"] = self.options.ray_masking
        self._cmake.definitions["EMBREE_BACKFACE_CULLING"] = self.options.backface_culling
        self._cmake.definitions["EMBREE_IGNORE_INVALID_RAYS"] = self.options.ignore_invalid_rays
        self._cmake.definitions["EMBREE_ISPC_SUPPORT"] = False
        self._cmake.definitions["EMBREE_TASKING_SYSTEM"] = "INTERNAL"

        self._cmake.configure(build_folder=self._build_folder)
        return self._cmake
```